modules/gapi/src/backends/ocl/goclcore.cpp

```diff
@@ -523,6 +541,47 @@ GAPI_OCL_KERNEL(GOCLTranspose, cv::gapi::core::GTranspose)
     }
 };
 
+GAPI_OCL_KERNEL(GOCLBGR, cv::gapi::streaming::GBGR)
+{
+    static void run(const cv::MediaFrame& in, cv::UMat& out)
+    {
+        (void)in; (void)out;
+#ifdef HAVE_DIRECTX
+#ifdef HAVE_D3D11
+#ifdef HAVE_ONEVPL
+        auto d = in.desc();
+        GAPI_Assert(d.fmt == cv::MediaFormat::NV12);
+
+        // FIXME: consider a better solution.
+        // Current approach cannot be easily extended for other adapters (getHandle).
+        auto params = in.get<cv::gapi::wip::onevpl::VPLMediaFrameDX11Adapter>()->getHandle();
+        auto handle = cv::util::any_cast<mfxHDLPair>(params);
+        ID3D11Texture2D* texture = reinterpret_cast<ID3D11Texture2D*>(handle.first);
+        GAPI_Assert(texture != nullptr && "mfxHDLPair contains ID3D11Texture2D that is nullptr");
```

Comment:
there are a lot of textures & surfaces and due to decoder/file/environment 'stability' there may be only one which is bad or cause additional issues.
I suggest we should log these data here, like https://github.com/opencv/opencv/blob/36f145da8f2aed61cf6482400500c96506a2ce5e/modules/gapi/src/streaming/onevpl/accelerators/surface/dx11_frame_adapter.cpp#L68

Then in case of failure we will "track" a bad surface through whole pipeline and find an source of issues

Reply:
Haven't managed to find a way to track properties of bad surfaces/textures/etc. Any recommendations where to look?

Reply:
just printing out handle as suggested above is enough. Later we can track surface by its pointer in DEBUG output and find out for which parameters this surface was created

Reply:
Done